main.py

```diff
@@ -0,0 +1,41 @@
+from typing import Optional
+
+from indexer.crawl_result import CrawlResult
+from indexer.database_connection import DatabaseConnectionParameters
+from indexer.indexer import Indexer
+from indexer.indexer_type import IndexerType
+from indexer.language import Language
+from indexer.rabbitmq_connection import RabbitMQConnectionParameters
+
+"""
+    Usage example
+"""
+
+if __name__ == '__main__':
+    class GitLabIndexer(Indexer):
+        def crawl_next_repository(self, prev_repository_id: Optional[str]) -> CrawlResult:
+            return CrawlResult(
+                               str(
+                                   int(prev_repository_id if prev_repository_id is not None else 7)+1
+                               ),
+                               "https://url",
+                               "git://url",
+                               {
+                                   Language.C: True,
+                                   Language.CPP: False,
+                                   Language.CSHARP: False,
+                                   Language.CSS: True,
+                                   Language.JAVA: True,
+                                   Language.JS: False,
+                                   Language.PHP: False,
+                                   Language.Python: False,
+                                   Language.Ruby: True
+                                }
+                               )
+
+    rabbit = RabbitMQConnectionParameters("172.17.0.2")
+    database = DatabaseConnectionParameters("172.17.0.3", "frege", "postgres", "password")
+
+    app = GitLabIndexer(IndexerType.GITLAB, rabbit, database, 10)
+
+    app.run()
```

Comment:
Simple (and fake) indexer example - in general usage example.

Mainly, we must implement a `crawl_next_repository` method from `Indexer` abstract class. This method should crawl the next repository and return a `CrawlResult` data-class filled with proper values (according to a [documentation](https://teams.microsoft.com/_#/docx/viewer/teams/https:~2F~2Fujchmura.sharepoint.com~2Fteams~2FOtwarterepozytoriakoduipomiaroprogramowania~2FShared%20Documents~2FGeneral~2Fspecyfikacja%20mikroserwis%C3%B3w.docx?threadId=19:18b7a43caae64e188110113719838c77@thread.tacv2&baseUrl=https:~2F~2Fujchmura.sharepoint.com~2Fteams~2FOtwarterepozytoriakoduipomiaroprogramowania&fileId=8e48bc23-26d3-4158-863f-9f6507f8862b&ctx=files&rootContext=items_view&viewerAction=view)).

Next, we must fill up a `RabbitMQConnectionParameters` and `DatabaseConnectionParameters` structures according to environment vars passed to our container, create our class instance and invoke a `run` method in it.